src/Wolf.cpp

```diff
@@ -0,0 +1,466 @@
+  uint atomSize = thisKind.NumAtoms();
+  uint start = mols.MolStart(molIndex);
+  double lambdaCoef = GetLambdaCoef(molIndex, box);
+
+  for (uint i = 0; i < atomSize; i++) {
+    if(particleHasNoCharge[start + i]) {
+      continue;
+    }
+    for (uint j = i + 1; j < atomSize; j++) {
+      if(currentAxes.InRcut(distSq, virComponents, trialMol.GetCoords(),
+                         i, j, box)){
+        // We eliminate the dampened rij term by choosing psi = 1
+        correction -= thisKind.AtomCharge(i) * thisKind.AtomCharge(j) *
+                      wolfFactor1[box];
+      }
+    }
+  }
+  GOMC_EVENT_STOP(1, GomcProfileEvent::CORR_SWAP);
+  return num::qqFact * correction * lambdaCoef * lambdaCoef;
+}
+
+//It's called in free energy calculation to calculate the change in
+// self energy in all lambda states
+void Wolf::ChangeSelf(Energy *energyDiff, Energy &dUdL_Coul,
+                         const std::vector<double> &lambda_Coul,
+                         const uint iState, const uint molIndex,
+                         const uint box) const
+{
+    uint lambdaSize = lambda_Coul.size();
+    double coefDiff, en_self = 0.0;
+    //Load the self energy with lambda = 1
+    en_self = molSelfEnergies[molIndex];
+    
+    // M_2_SQRTPI is 2/sqrt(PI), so need to multiply by 0.5 to get sqrt(PI)
+    //Vlugt
+    //en_self *= ((ff.wolfAlpha[box] * M_2_SQRTPI * 0.5) +  ff.wolfFactor1[box] );
+    // We eliminate the alpha/root(pi) using Wolf,mod
+    en_self *= -1.0 * ff.wolfFactor1[box] * num::qqFact;
+
+    //Calculate the energy difference for each lambda state
+    for (uint s = 0; s < lambdaSize; s++) {
+      coefDiff = lambda_Coul[s] - lambda_Coul[iState];
+      energyDiff[s].self += coefDiff * en_self;
+    }
+    //Calculate du/dl of self for current state, for linear scaling
+    dUdL_Coul.self += en_self;
+}
+
+//It's called in free energy calculation to calculate the change in
+// correction energy in all lambda states
+void Wolf::ChangeCorrection(Energy *energyDiff, Energy &dUdL_Coul,
+                               const std::vector<double> &lambda_Coul,
+                               const uint iState, const uint molIndex,
+                               const uint box) const
+{
+  uint atomSize = mols.GetKind(molIndex).NumAtoms();
+  uint start = mols.MolStart(molIndex);
+  uint lambdaSize = lambda_Coul.size();
+  double coefDiff, distSq, dist, correction = 0.0;
+  XYZ virComponents;
+
+  //Calculate the correction energy with lambda = 1
+  for (uint i = 0; i < atomSize; i++) {
+    if(particleHasNoCharge[start + i]) {
+      continue;
+    }
+
+    for (uint j = i + 1; j < atomSize; j++) {
+      distSq = 0.0;
+      if(currentAxes.InRcut(distSq, virComponents, currentCoords,
+                         start + i, start + j, box)){
+        // We eliminate the dampened rij term by choosing psi = 1
+        correction += particleCharge[i + start] * particleCharge[j + start] * wolfFactor1[box];
+      }
```

Comment:
The rcut that boxDim uses to check the distance is max(rcut, rcutcoulomb). You need to additionally check for rcutcoulomb, otherwise if user set a cutoff that `rcut > rcutcoulomb` the results will be wrong.